cpp/StrangeCharacters/test/ExampleCharacterTestCase.cpp

```diff
@@ -0,0 +1,40 @@
+#include "Character.hpp"
+#include "CharacterFinder.hpp"
+#include <catch2/catch.hpp>
+
+namespace Characters::Test {
+
+/*
+ * This test case is not part of the exercise - it is example code showing what's possible
+ */
+TEST_CASE("Find Character By Last Name") {
+    // This test constructs all its own test data to make it clearer what it does
+    auto karen = std::make_unique<Character>("Karen", "Wheeler");
+    auto mike = std::make_unique<Character>("Mike", "Wheeler");
+
+    // this is example code showing kinds of assertion you could do on a Character
+    auto nancy = std::make_unique<Character>("Nancy", "Wheeler");
+    REQUIRE("Nancy" == nancy->FirstName);
+    REQUIRE(nancy == nancy);
+
+    // This assertion works because Character has a custom operator==
+    REQUIRE(*std::make_unique<Character>("Nancy", "Wheeler") == *nancy);
+
+    karen->AddChild(nancy.get());
+    karen->AddChild(mike.get());
+
+    std::vector<std::unique_ptr<Character>> characters;
+    characters.push_back(std::move(nancy));
+    characters.push_back(std::move(mike));
+    characters.push_back(std::move(karen));
+
+    CharacterFinder finder(std::move(characters));
+
+    // This is the 'act' step
+    auto charactersList = finder.FindFamilyByLastName("Wheeler");
+    
+    // this is example code showing kinds of assertion you could do on a List of Characters
+    REQUIRE(3 == charactersList.size());
```

Comment:
this is a good start, would like some more kinds of assertion here

Reply:
specificlly, some of these matchers checking the contents of the charactersList http://google.github.io/googletest/reference/matchers.html

Reply:
So far the test cases use Catch2, which has a nicer syntax (compare `REQUIRE(3 == charactersList.size())` to `EXPECT_EQ(3, charactersList.size())`). What kind of assertions did you want? Some of the ones mentioned in C# don't make sense in C++.

Reply:
The thing I'm bumping up against is that the C# code has a, for C++ interpretation, loose handling between "object is equal" and "handle to object is equal". Originally I wrote the code to have the Character as a vector, making them actually live in that vector. That makes the vector unmoveable (since they'd move, and the pointers would dangle), so I changed it to use unique_ptr to hold them instead.

That now makes unique_ptr<Character> not equal to another unique_ptr<Character> since they're address-compared by default.

I'll see what I can do.

Reply:
This is exactly why I needed someone who actually knows C++ properly to work on this! My understanding of a ValueObject is that if the contents are the same then they are the same, even if they are different objects with different handles. They are of course also equal if you compare two handles and they point to the same object. I think the problem you point out is that you have two objects that are the same but it doesn't get as far as comparing the objects because the handles are different. If a Character were a string then I think you would get the behaviour I'm looking for. Can you achieve that with a user defined class in C++?

So the point of the assertions in C# is that I construct a new list with new objects in that have different handles but the same data as the originals. That makes my assertions more readable. That's really what I'm after - how do you write the assertions in the best way so it's really clear which Characters I'm expecting in the result? So perhaps the whole discussion on handles is misplaced - how would you write really good assertions for this in C++ and with your chosen test framework?